samza-kv/src/main/scala/org/apache/samza/storage/kv/KeyValueStorageEngine.scala

```diff
@@ -160,4 +160,11 @@ class KeyValueStorageEngine[K, V](
   }
 
   override def getStoreProperties: StoreProperties = storeProperties
+
+  override def iterate(from: K, to: K): KeyValueIterable[K, V] = {
+    updateTimer(metrics.iterateNs) {
```

Comment:
not sure how this works..

Reply:
Scala magic :). It's called currying, I finally remembered the name of it :)